backend/infrahub/core/node/__init__.py

```diff
@@ -100,6 +100,13 @@ def get_id(self) -> str:
     def get_updated_at(self) -> Timestamp | None:
         return self._updated_at
 
+    def uses_profiles(self) -> bool:
+        for attr_name in self.get_schema().attribute_names:
+            node_attr = getattr(self, attr_name, None)
```

Comment:
```diff
- node_attr = getattr(self, attr_name, None)
+ node_attr = self.get_schema().get_attribute(name=attr_name)
```
This would be cleaner from a typing point of view. It would raise an error if we tried to get an attribute that didn't exist but given the context that should never happen.